source/event.h

```diff
@@ -20,12 +20,19 @@
 extern "C" {
 #endif
 
+/* We borrow base event storage, including its list link, and add a time field */
```

Comment:
Shouldn't this be in the timer's private header? Alternatively, remove the need to expose it to event.c by having an internal "timer_sys_event_cancel_period" (can't think of a good name).

Or maybe timer_sys_event_cancel_critical could do the job, and we restructure this a bit? Always set period to 0, and remove from its own list if event state is UNQUEUED?